rust/src/buffer.rs

```diff
@@ -41,24 +42,27 @@ impl<T> Buffer<T> {
         self.len
     }
 
+    /// Get a pointer to the data contained by the buffer
     pub fn data(&self) -> *const T {
         self.data
     }
 
     pub fn slice(&self, start: usize, end: usize) -> &[T] {
         assert!(start <= end);
-        assert!(start <= self.len as usize);
+        assert!(start < self.len as usize);
         assert!(end <= self.len as usize);
         unsafe { slice::from_raw_parts(self.data.offset(start as isize), (end - start) as usize) }
     }
 
     /// Get a reference to the value at the specified offset
     pub fn get(&self, i: usize) -> &T {
```

Comment:
Good idea to combine `slice` and `get`, to mirror the stdlib [slice api](https://doc.rust-lang.org/std/primitive.slice.html#method.get) (it returns a slice if passed a `Range`, item if passed a `usize`. Depending on how `Buffer` is used, might also be worth implementing `Index` and `IndexMut`